### ISSUE TITLE
`__mro_entries__` documented in PEP 560 but no supported

### ISSUE THREAD

**Issue by GreyElaina** [microsoft/pyright#6780]
**Describe the bug**
```python
class N:
    def __mro_entries__(self, bases):
        return (str, )

TypedDict = N()

class Test(N()):  # encounters reportGeneralTypeIssues
    a: str

Test.__mro__  # Us actually works as expected and documented in PEP 560
```

**VS Code extension or command-line**
[playground](https://pyright-play.net/?code=MYGwhgzhAECqEC4BQ1XQCYFMBm0D6eAtgE4D2emAdgC7ECWmEBAFBJiNgDTQBGkjASmRoR0YpmoBXYpWita3AUiQAVAJ4AHTOgAidYNWgBeOBGZKkoSDBWNqzeOaEo0YBNAi1ltzwDoCJOR4QA)

**Comment by erictraut** [comment]
This is an internal implementation detail used to implement `GenericAlias`, which is itself an internal implementation detail for supporting generic types in the runtime. It's not something that you should be using in your own code. If you do, you'll need to deal with the implications from the perspective of static type checking. I don't plan to add any support for this in pyright. Even if I wanted to add support for this in pyright, it's too dynamic for a static type checker to reason about. 

Not surprisingly, mypy also doesn't support this.
